engine/servermaster/jobmanager_test.go

```diff
@@ -712,12 +715,14 @@ func TestJobManagerIterPendingJobs(t *testing.T) {
 	mgr.JobFsm.JobOffline(mockHandle, true /* needFailover */)
 
 	// job is being backoff
+	mockJobOperator.EXPECT().IsJobCanceling(ctx, jobID).Times(1).Return(false)
 	mockBackoffMgr.EXPECT().Terminate(jobID).Times(1).Return(false)
 	mockBackoffMgr.EXPECT().Allow(jobID).Times(1).Return(false)
 	err = mgr.Tick(ctx)
 	require.NoError(t, err)
 
 	// job will be terminated
+	mockJobOperator.EXPECT().IsJobCanceling(ctx, jobID).Times(1).Return(false)
```

Comment:
It seems the test does not cover the `true` branch.

Reply:
fixed, add one more test.